docs/spire.md

```diff
@@ -0,0 +1,284 @@
+<!--
+---
+linkTitle: "SPIRE"
```

Comment:
NIT: I think "TaskRun result attestation" could be a better title - SPIRE is the technology we decided to use, but it does not really describe the feature

Reply:
Agreed. I will change it.